js/ui/date_range_box.d.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  EventInfo,
+  NativeEventInfo,
+  InitializedEventInfo,
+  ChangedOptionInfo,
+} from '../events/index';
+
+import {
+  DropDownButtonTemplateDataModel,
+} from './drop_down_editor/ui.drop_down_editor';
+
+import dxDateBox, { dxDateBoxOptions } from './date_box';
+
+import {
+  ValueChangedInfo,
+} from './editor/editor';
+
+/** @public */
+export type ChangeEvent = NativeEventInfo<dxDateRangeBox>;
+
+/** @public */
+export type ClosedEvent = EventInfo<dxDateRangeBox>;
+
+/** @public */
+export type ContentReadyEvent = EventInfo<dxDateRangeBox>;
+
+/** @public */
+export type CopyEvent = NativeEventInfo<dxDateRangeBox, ClipboardEvent>;
+
+/** @public */
+export type CutEvent = NativeEventInfo<dxDateRangeBox, ClipboardEvent>;
+
+/** @public */
+export type DisposingEvent = EventInfo<dxDateRangeBox>;
+
+/** @public */
+export type EnterKeyEvent = NativeEventInfo<dxDateRangeBox, KeyboardEvent>;
+
+/** @public */
+export type FocusInEvent = NativeEventInfo<dxDateRangeBox, FocusEvent>;
+
+/** @public */
+export type FocusOutEvent = NativeEventInfo<dxDateRangeBox, FocusEvent>;
+
+/** @public */
+export type InitializedEvent = InitializedEventInfo<dxDateRangeBox>;
+
+/** @public */
+export type InputEvent = NativeEventInfo<dxDateRangeBox, UIEvent & { target: HTMLInputElement }>;
+
+/** @public */
+export type KeyDownEvent = NativeEventInfo<dxDateRangeBox, KeyboardEvent>;
+
+/** @public */
+export type KeyPressEvent = NativeEventInfo<dxDateRangeBox, KeyboardEvent>;
+
+/** @public */
+export type KeyUpEvent = NativeEventInfo<dxDateRangeBox, KeyboardEvent>;
+
+/** @public */
+export type OpenedEvent = EventInfo<dxDateRangeBox>;
+
+/** @public */
+export type OptionChangedEvent = EventInfo<dxDateRangeBox> & ChangedOptionInfo;
+
+/** @public */
+export type PasteEvent = NativeEventInfo<dxDateRangeBox, ClipboardEvent>;
+
+/** @public */
+export type ValueChangedEvent =
+    NativeEventInfo<dxDateRangeBox, KeyboardEvent | MouseEvent | PointerEvent | Event>
+    & ValueChangedInfo;
+
+/** @public */
+export type DropDownButtonTemplateData = DropDownButtonTemplateDataModel;
+
+/**
+ * @deprecated use Properties instead
+ * @namespace DevExpress.ui
+ */
+export interface dxDateRangeBoxOptions extends Omit<dxDateBoxOptions<dxDateRangeBox>, 'value'> {
+  /**
+     * @docid
+     * @default null
+     * @public
+     */
+  endDate?: Date | number | string;
+  /**
+     * @docid
+     * @default 'calendar'
+     * @default 'native' &for(iOS)
+     * @default 'native' &for(Android)
+     * @public
+     */
+  pickerType?: 'calendar' | 'native';
+  /**
+     * @docid
+     * @default false
+     * @public
```

Comment:
shouldn't it be hidden?

Reply:
it's hidden but ins .js file

Reply:
Why we need to declare showAnalogClock there and then hide it?

Reply:
It's pretty common situation when we want hide option from docs but set another default value (for example tooltip: https://github.com/DevExpress/DevExtreme/blob/23_1/js/ui/tooltip/tooltip.js#L13)

In this case I thought that it's a bit strange to leave default value as true. But anyway option changes nothing, because there is no `datetime` mode in DateRangeBox, so clock will never be shown, so I think I'll get rid of this defalut value change